shakedown/dcos/__init__.py

```diff
@@ -162,3 +167,125 @@ def _gen_url(url_path):
     """
     from six.moves import urllib
     return urllib.parse.urljoin(dcos_url(), url_path)
+
+
+@lru_cache(maxsize=None)
+def _get_connection(host, username, key_path):
+    """Return an authenticated SSH connection.
+
+    :param host: host or IP of the machine
+    :type host: str
+    :param username: SSH username
+    :type username: str
+    :param key_path: path to the SSH private key for SSH auth
+    :type key_path: str
+    :return: SSH connection
+    """
+    if not username:
+        username = shakedown.cli.ssh_user
+    if not key_path:
+        key_path = shakedown.cli.ssh_key_file
+    key = validate_key(key_path)
+    transport = get_transport(host, username, key)
+    
+    if transport:
+        transport = start_transport(transport, username, key)
+        if transport.is_authenticated():
+            return transport
+        else:
+            print("error: unable to authenticate {}@{} with key {}".format(username, host, key_path))
+    else:
+        print("error: unable to connect to {}".format(host))
+    
+    return None
+
+
+def get_session(host, username, key_path):
+    """Return a new session on an authenticated SSH connection.
+
+    :param host: host or IP of the machine
+    :type host: str
+    :param username: SSH username
+    :type username: str
+    :param key_path: path to the SSH private key for SSH auth
+    :type key_path: str
+    :return: SSH connection
+    """
+    c = _get_connection(host, username, key_path)
+    return c.open_session()
+
+
+class HostSession:
```

Comment:
wouldn't it be better to annotate with `@contextmanager`
```
from contextlib import contextmanager
@contextmanager
```

Reply:
Can I use that on a `class`? I use the class strategy to provide an abstraction (`run`) over the SSH session.